pennylane/transforms/control.py

```diff
@@ -0,0 +1,191 @@
+        # TODO(chase): Do we need to re-queue these ops?
+        return tape
+
+    def adjoint(self, do_queue=False):
+        def requeue_tape(tape):
+            for op in tape.operations:
+                op.queue()
+
+        with QuantumTape(do_queue=False) as new_tape:
+            # Execute all ops adjointed.
+            adjoint(requeue_tape)(self.subtape)
+        return ControlledOperation(new_tape, self.control_wires, do_queue=do_queue)
+
+
+register_control(
+    ControlledOperation,
+    lambda op, wires: ControlledOperation(
+        tape=op.subtape,
+        control_wires=Wires(wires) + op.control_wires,
+    ),
+)
+
+
+def ctrl(fn, control):
+    """Create a method that applies a controlled version of the provided method.
+
+    Args:
+        fn (function): Any python function that applies pennylane operations.
+        control (Wires): The control wire(s).
+
+    Returns:
+        A new function that applies the controlled equivalent of `fn`.
+
+    **Example**
+
+    .. code-block:: python3
+
+        def ops(params):
+            qml.RX(params[0], wires=0)
+            qml.RZ(params[1] wires=3)
+
+        ops1 = qml.ctrl(ops, control=1)
+        ops2 = qml.ctrl(ops, control=2)
+
+        @qml.qnode(dev)
+        def my_circuit():
+            ops1(params=[0.123, 0.456])
+            ops1(params=[0.789, 1.234])
+            ops2(params=[2.987, 3.654])
+            ops2(params=[2.321, 1.111])
+            return qml.state()
+
+    The above code would be equivalent to
+
+    .. code-block:: python3
+
+        @qml.qnode(dev)
+        def my_circuit(params):
+            # ops1(params=[0.123, 0.456])
+            qml.CRX(0.123, wires=[1, 0])
+            qml.CRZ(0.456, wires=[1, 3])
+
+            # ops1(params=[0.789, 1.234])
+            qml.CRX(0.789, wires=[1, 0])
+            qml.CRZ(1.234, (wires=[1, 3])
+
+            # ops2(params=[2.987, 3.654])
+            qml.CRX(2.987, wires=[2, 0])
+            qml.CRZ(3.654, wires=[2, 3])
+
+            # ops2(params=[2.321, 1.111])
+            qml.CRX(2.321, wires=[2, 0])
+            qml.CRZ(1.111, wires=[2, 3])
+            return qml.state()
+
+    .. Note::
+
+        Some devices are able to take advantage of the inherient sparsity of a
+        controlled operation. In those cases, it may be more efficient to use
+        this transform rather than adding controls by hand. For devices that don't
+        have special control support, the operation is expanded to add control wires
+        to each underlying op individually.
+
+    .. UsageDetails::
+
+        **Nesting Controls**
+
+        The ``ctrl`` transform can be nested with itself arbitrarily.
+
+        .. code-block:: python3
+
+            # These two ops are equivalent.
+            op1 = ctrl(ctrl(my_ops, 1), 2)
+            op2 = ctrl(my_ops, [2, 1])
+    """
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        with QuantumTape(do_queue=False) as tape:

```

Comment:
I am so confused about the entire `do_queue` thing by now ;)

Reply:
`do_queue=False` prevents this QuantumTape from being queued in a parent take context. We need that here (and frankly, all tape transforms) to prevent our scratch work from also getting queued. 

If `do_queue` was `True` here, all of the ops execute in `fn` would then be executed both with and without controls.